### ISSUE TITLE
Integrate with "Mijn lijst"

### ISSUE THREAD

**Issue by michaelarnauts** [michaelarnauts/plugin.video.vtm.go#47]
VTM go has a "Mijn lijst" to add your favorite shows or movies.

The program json contains a `program.addedToMyList` boolean. That indicates if it is on "Mijn lijst".

# To get your list
* GET https://api.vtmgo.be/vtmgo/main/swimlane/my-list
```
{
    "id": "my-list",
    "logoUrl": null,
    "metaData": {
        "abGroup": null,
        "provider": "vtmgo",
        "requestId": null,
        "routingGroup": null
    },
    "rowType": "MY_LIST",
    "teasers": [
        {
            "geoBlocked": false,
            "imageUrl": "https://images4.persgroep.net/rcs/YVEYXMJGh6vITW7JSBnFTVspK0Y/diocontent/143701390/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "primaryImageUrl": "https://images4.persgroep.net/rcs/YVEYXMJGh6vITW7JSBnFTVspK0Y/diocontent/143701390/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "target": {
                "id": "9f40aea2-3a9a-48f9-b57a-d10495a12995",
                "name": "13 Geboden",
                "type": "PROGRAM"
            },
            "title": "13 Geboden",
            "type": "SCREEN"
        },
        {
            "geoBlocked": false,
            "imageUrl": "https://images1.persgroep.net/rcs/HJ3F6neo1irh05oHHUa2o4J73_E/diocontent/152777694/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "primaryImageUrl": "https://images1.persgroep.net/rcs/HJ3F6neo1irh05oHHUa2o4J73_E/diocontent/152777694/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "target": {
                "assetType": "movie",
                "contextIdType": "video",
                "id": "9686e7ec-8084-4b29-98a4-3896aefa9f95",
                "name": "Baby Sellers",
                "playableId": "9686e7ec-8084-4b29-98a4-3896aefa9f95",
                "serviceType": "context",
                "type": "MOVIE"
            },
            "title": "Baby Sellers",
            "type": "SCREEN"
        },
        {
            "geoBlocked": false,
            "imageUrl": "https://images0.persgroep.net/rcs/LeK3LDlHX-GXU6dnhvyCdbw1sXs/diocontent/153922961/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "primaryImageUrl": "https://images0.persgroep.net/rcs/LeK3LDlHX-GXU6dnhvyCdbw1sXs/diocontent/153922961/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "target": {
                "assetType": "movie",
                "contextIdType": "video",
                "id": "68549907-6532-45a8-a62d-c57e60b82ea2",
                "name": "Bad Boys II",
                "playableId": "68549907-6532-45a8-a62d-c57e60b82ea2",
                "serviceType": "context",
                "type": "MOVIE"
            },
            "title": "Bad Boys II",
            "type": "SCREEN"
        },
        {
            "geoBlocked": false,
            "imageUrl": "https://images0.persgroep.net/rcs/UM0taEagdSUvQ3OL6sY1ob9NRLg/diocontent/144640991/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "primaryImageUrl": "https://images0.persgroep.net/rcs/UM0taEagdSUvQ3OL6sY1ob9NRLg/diocontent/144640991/_fitwidth/426?appId=038a353bad43ac27fd436dc5419c256b&quality=0.8",
            "target": {
                "id": "3561ebdd-d9ce-41ad-a0ae-420459c94748",
                "name": "Wat Als",
                "type": "PROGRAM"
            },
            "title": "Wat Als",
            "type": "SCREEN"
        }
    ],
    "title": "Mijn Lijst"
}
```

# To add an item
* PUT https://api.vtmgo.be/vtmgo/userData/myList/PROGRAM/3b2209e9-4fb3-4516-aa57-47edbafcdeb4
* PUT https://api.vtmgo.be/vtmgo/userData/myList/MOVIE/9686e7ec-8084-4b29-98a4-3896aefa9f95

The api replies with a 204 without content.

# To remove an item
* DELETE https://api.vtmgo.be/vtmgo/userData/myList/PROGRAM/3b2209e9-4fb3-4516-aa57-47edbafcdeb4

The api replies with a 204 without content.

We need to specify the `x-dpp-jwt` token and include the user jwt in all api's.

**Comment by dagwieers** [comment]
I think it would be useful to document the various VTM GO APIs in the Wiki.

**Comment by michaelarnauts** [comment]
The management of the My list isn't done yet. There is no way to know in a listing of an item is added on "my list".
We could just add an add to my list anyway, and only allow removing from my list on the "my list" section itself.

**Comment by michaelarnauts** [comment]
Fixed in d535e05e604d7ceba81b768e27075dae5a1694e4
